eos/utils/observable_cache.cc

```diff
@@ -83,13 +85,50 @@ namespace eos
     void
     ObservableCache::update()
     {
+#ifdef EOS_ENABLE_PMC
+#else
         // evaluate all observables
+        std::vector<Ticket> tickets;
+        tickets.reserve(_imp->predictions.size());
+#endif
+
         auto p = _imp->predictions.begin();
 
         for (auto o = _imp->observables.begin(), o_end = _imp->observables.end() ; o != o_end ; ++o, ++p)
         {
+#ifdef EOS_ENABLE_PMC
+            // evaluate observables one by one
             *p = (*o)->evaluate();
+#else
+            // parallelize the evaluation of the observables
+            // since the EOS builtin PMC sampler is parallelized,
+            // parallelizing the ObservableCache can and will lead
+            // to a deadlock.
+            auto f = [=]() {
+                try
+                {
+                    *p = (*o)->evaluate();
+                }
+                catch (eos::Exception & e)
+                {
+                    Log::instance()->message("ObservableCache::update", ll_error)
+                        << "Exception encountered when evaluating observable '" << (*o)->name() << "[" << (*o)->kinematics().as_string() << "];" << (*o)->options().as_string() << "': "
+                        << e.what();
+                    *p = std::numeric_limits<double>::quiet_NaN();
+
+                }
+            };
```

Comment:
Good point. So far, protection was not necessary. Exceptions were caught downstream in the individual workers. With parallelization in the ObservableCache, we need to catch exceptions now here. Once PMC is entirely removed (long term goal!), then the piece without a try/catch around it will be removed anyway.